math/minuit2/src/MnSeedGenerator.cxx

```diff
@@ -160,13 +159,13 @@ MinimumSeed MnSeedGenerator::operator()(const MnFcn &fcn, const AnalyticalGradie
 
    NegativeG2LineSearch ng2ls;
    if (ng2ls.HasNegativeG2(dgrad, prec)) {
-      Numerical2PGradientCalculator ngc(fcn, st.Trafo(), stra);
-      state = ng2ls(fcn, state, ngc, prec);
+      state = ng2ls(fcn, state, gc, prec);
```

Comment:
I am not sure that removing the usage of the Numerical2PGradientCalculator is ok for all cases.
The following function , NegativeG2LineSearch::operator(), uses the G2 and GStep. If we have an analytical gradient not implementing these it can be problematic. 
Actually why not leaving it always in all cases ?

Reply:
I have put it back in and it seems to have no effect on my tests, thanks for the suggestion.